### ISSUE TITLE
你好！有一行代码不知道对应论文中的哪个公式想要请教一下

### ISSUE THREAD

**Issue by hangjiaqi1** [thunlp/GP-GNN#1]
在our_models.py的107行中，您定义# rnn_output shape: batch * MAX_EDGES_PER_GRAPH, max_sent_len, hidden
我设置的是batch_size = 10 所以size = (720, 36, 512)
108行代码
rnn_result = torch.cat([rnn_output[:, -1, :self.p['units1']], rnn_output[:, 0, self.p['units1']:]], dim=1).view(sentence_input.size()[0], MAX_EDGES_PER_GRAPH, -1)
我的理解是： 在108行中， 相当于截取lstm中第35个word的前0到255的embedding，和第0个word的第256到512的embedding。经过tocrh,cat变成维度（720，2，256）最后再view一下变成（10， 72，512 ）,view相当于把同一个sentence中的第一个word和最后1个word的各自256维度拼接。这个第一个词后半embedding后最后一个词的前半embedding的拼接。作为graph edge的value
我的问题：如果我理解的没有错，请问这样做有什么依据吗？它对应了论文中哪一条公式？

**Comment by xcjthu** [comment]
关于第一个词和最后一个词的选取，这是和双向的RNN有关的，选取双向RNN的两个方向的最后一个隐向量来作为边的表示。

**Comment by bobobe** [comment]
Thanks，I'll reply you soon

**Comment by LOUGBERTI** [comment]
> 在our_models.py的107行中，您定义# rnn_output shape: batch * MAX_EDGES_PER_GRAPH, max_sent_len, hidden 我设置的是batch_size = 10 所以size = (720, 36, 512) 108行代码 rnn_result = torch.cat([rnn_output[:, -1, :self.p['units1']], rnn_output[:, 0, self.p['units1']:]], dim=1).view(sentence_input.size()[0], MAX_EDGES_PER_GRAPH, -1) 我的理解是： 在108行中， 相当于截取lstm中第35个word的前0到255的embedding，和第0个word的第256到512的embedding。经过tocrh,cat变成维度（720，2，256）最后再view一下变成（10， 72，512 ）,view相当于把同一个sentence中的第一个word和最后1个word的各自256维度拼接。这个第一个词后半embedding后最后一个词的前半embedding的拼接。作为graph edge的value 我的问题：如果我理解的没有错，请问这样做有什么依据吗？它对应了论文中哪一条公式？

您好，请问您有实现P@K这个指标吗？我并没有找到这个指标的代码
